image_classification/main.js

```diff
@@ -154,7 +248,31 @@ async function renderCamStream() {
 
 // Get top 3 classes of labels from output buffer
 function getTopClasses(buffer, labels) {
-  const probs = Array.from(buffer);
+  let float32Buffer = buffer;
+  // Currently we need to fallback softmax to tf.softmax because
+  // NPU dosen't support softmax.
+  // Convert output buffer from float16 to float32, because
+  // tf.tensor/tf.softmax doesn't support float16 data type
+  // according to https://js.tensorflow.org/api/latest/#tensor.
+  // TODO: Remove this workaround once NPU supports softmax.
+  if (inputOptions.dataType === 'float16') {
+    const elementsCount = utils.sizeOfShape(netInstance.outputDimensions);
+    const float32Array = new Float32Array(elementsCount);
+    for (let i = 0; i < elementsCount; ++i) {
+      float32Array[i] = utils.float16ToNumber(buffer[i]);
```

Comment:
Converting float32 to float16 is slow, please try to use WebNN cast operator.

Reply:
Done, thanks!